pkg/types/license.go

```diff
@@ -0,0 +1,177 @@
+package types
+
+import (
+	"context"
+	"errors"
+
+	"github.com/kong/go-database-reconciler/pkg/crud"
+	"github.com/kong/go-database-reconciler/pkg/state"
+	"github.com/kong/go-kong/kong"
+)
+
+type licenseCRUD struct {
+	client *kong.Client
+	// isKonnect indicates whether it is syncing with Konnect and licenses should not get changed.
+	isKonnect bool
+}
+
+var _ crud.Actions = &licenseCRUD{}
+
+func licenseFromEventStruct(arg crud.Event) *state.License {
+	license, ok := arg.Obj.(*state.License)
+	if !ok {
+		panic("unexpected type, expected *state.License")
+	}
+	return license
+}
+
+// Create creates a License in Kong.
+// The arg should be of type crud.Event, containing the license to be created,
+// else the function will panic.
+// It returns a the created *state.Licen
+func (s *licenseCRUD) Create(ctx context.Context, arg ...crud.Arg) (crud.Arg, error) {
+	if s.isKonnect {
+		return nil, nil
+	}
+	event := crud.EventFromArg(arg[0])
+	license := licenseFromEventStruct(event)
+	createdLicense, err := s.client.Licenses.Create(ctx, &license.License)
+	if err != nil {
+		return nil, err
+	}
+	return &state.License{License: *createdLicense}, nil
+}
+
+// Delete deletes a License in Kong.
+// The arg should be of type crud.Event, containing the license to be deleted,
+// else the function will panic.
+// It returns a the deleted *state.License.
+func (s *licenseCRUD) Delete(ctx context.Context, arg ...crud.Arg) (crud.Arg, error) {
+	if s.isKonnect {
+		return nil, nil
+	}
+	event := crud.EventFromArg(arg[0])
```

Comment:
Can `arg` be not provided? This will `panic` in that case. WDYT about checking `len(arg)`?

Ditto for all the other use cases above.

Reply:
This is just the same as the CRUD functions for other types. I think it is OK for checking `len(arg)` and returning errors but this needs some discussion about how to change the existing CRUD functions for the other types.

Reply:
Added errors for the case.